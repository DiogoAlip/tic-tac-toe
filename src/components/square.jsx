export const Square = ({ children, isSelected, updateBoard, index }) => {
	const className = `square ${isSelected ? "is-selected" : ""}`;
	return (
		// rome-ignore lint/a11y/useKeyWithClickEvents: <explanation>
		<div data-testid="Square-id" onClick={updateBoard} className={className}>
			{children}
		</div>
	);
};
