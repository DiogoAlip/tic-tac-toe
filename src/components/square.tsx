import React from "react";

type Props = {
  children?: JSX.Element | number;
  isSelected?: boolean;
  updateBoard?: () => void;
  index?: number;
};

export const Square = ({ children, isSelected, updateBoard, index }: Props) => {
  const className = `square ${isSelected ? "is-selected" : ""}`;
  return (
    <React.Fragment>
      {/** rome-ignore lint/a11y/useKeyWithClickEvents: <explanation>*/}
      <div data-testid="Square-id" onClick={updateBoard} className={className}>
        {children}
      </div>
    </React.Fragment>
  );
};
