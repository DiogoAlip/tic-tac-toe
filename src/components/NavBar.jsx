import { Square } from "./square";
import { marks } from "../constans";
import { useState } from "react";

export function NavBar({board, setBoard, turn, TURNS}){
    const [modal, setModal] = useState(false)
    const activateModal = () => {
        setModal(!modal)
    }

    const simbolChanger = (number) => {
        const newBoard = board.map((element)=>{
            if(element == turn){element = marks[number]}
            
        })
        setBoard(newBoard)

        for( let i in TURNS){
            if(i == turn){
                i = marks[number]
            }
            console.log(i)
        }
        activateModal()
    }

    return <div className="nav">
        <button onClick={activateModal}>Super Mishi</button>
        {modal && <section className="character-selector">
            {
                marks.map((_,index) => {
                    return (<Square key={index}>
                        <img 
                            className="table_element"
                            src={marks[index]}
                            alt=""
                            onClick={simbolChanger}/>
                    </Square>)
                })
            }
        </section>}
    </div>
}