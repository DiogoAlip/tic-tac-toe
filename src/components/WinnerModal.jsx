import { Square } from "./square"
export const WinnerModal = ({winner, resetGame}) => {
    if (winner == null) return null

    const TextWinnner =  winner == false ? 'Empate' : 'Gano' 

    return (
        <section className="winner">
            <div className="text">
                <h2>
                    {TextWinnner}
                </h2>
    
                <header className="win">
                    {winner && 
                    <Square><img className="table_element" src={winner} alt="" /></Square>}
                </header>
    
                <footer>
                    <button onClick={resetGame}>Empezar de nuevo</button>
                </footer>
            </div>
        </section>
    )
}