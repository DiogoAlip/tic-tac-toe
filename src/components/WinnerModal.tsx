import { Square } from "./square";
export const WinnerModal = ({
  winner,
  resetGame,
}: {
  winner: null | React.FC<React.SVGProps<SVGSVGElement>>;
  resetGame: () => void;
}) => {
  if (winner == null) return null;

  const TextWinnner = winner === null ? "Empate" : "Gano";

  return (
    <section className="winner">
      <div className="text">
        <h2>{TextWinnner}</h2>

        <header className="win">
          {winner && (
            <Square>
              <img className="table_element" src={winner.toString()} alt="" />
            </Square>
          )}
        </header>

        <footer>
          <button type="button" onClick={resetGame}>
            Empezar de nuevo
          </button>
        </footer>
      </div>
    </section>
  );
};
