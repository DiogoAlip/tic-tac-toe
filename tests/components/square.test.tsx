import { fireEvent, render, screen } from "@testing-library/react";
import { Square } from "../../src/components/square";

describe("square.jsx", () => {
  const string = "string";

  test("Square should match with the snapshot", () => {
    const { container } = render(<Square />);
    expect(container).toMatchSnapshot();
  });

  test("Square should return a children", () => {
    const headear1 = <h1>{string}</h1>;
    render(
      <Square>
        <p>{string}</p>
      </Square>
    );
    expect(screen.getByText(string).innerHTML).toBe(string);

    render(<Square>{headear1}</Square>);
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toBe(string);
  });

  test("Square shouldnt return className", () => {
    const name = "square ";

    const { container } = render(<Square />);
    expect(screen.getByTestId("Square-id")).toBeTruthy();
    expect(container.getElementsByClassName(`${name}`)).toBeTruthy();
  });

  test("Square should return className", () => {
    const selectIsTrue = true;
    const name = "square is-selected";

    const { container } = render(<Square isSelected={true} />);
    expect(screen.getByTestId("Square-id")).toBeTruthy();
    expect(container.getElementsByClassName(`${name}`)).toBeTruthy();
  });

  test("Square should return an event", () => {
    let variable = false;
    render(
      <Square
        updateBoard={() => {
          variable = true;
        }}
      />
    );
    fireEvent.click(screen.getByTestId("Square-id"));
    expect(variable).toBeTruthy();
  });
});
