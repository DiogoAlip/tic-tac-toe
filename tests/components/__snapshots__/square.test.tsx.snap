// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`square.jsx Square should match with the snapshot 1`] = `
<div>
  <div
    class="square "
    data-testid="Square-id"
  />
</div>
`;
